### PR TITLE
Add CBMC proof of requestJobMqtt function

### PR DESCRIPTION
**Pull request by Bujain**
Add CBMC proof of requestJobMqtt function

Description
-----------
Add CBMC proof files of requestJobMqtt function

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] My code is formatted using Uncrustify.
- [x] I have read and applied the rules stated in CONTRIBUTING.md. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/cbmc/proofs/requestJob_Mqtt/Makefile [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+# This should be a unique identifier for this proof, and will appear on the
+# Litani dashboard. It can be human-readable and contain spaces if you wish.
+PROOF_UID = requestJob_Mqtt
+
+DEFINES +=
+INCLUDES +=
+
+REMOVE_FUNCTION_BODY +=
+UNWINDSET +=
+
```
Comment (divekarshubham) [inline test/cbmc/proofs/requestJob_Mqtt/Makefile:16]: Please remove unused variables.

File: test/cbmc/proofs/requestJob_Mqtt/requestJob_Mqtt_harness.c [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    OtaAgentContext_t agent;
+    OtaInterfaces_t otaInterface;
+
+    /* publish reference to the mqtt function is expected to be assigned by the user and thus
+     * assumed not to be NULL. */
+    mqtt.publish = publish;
+    otaInterface.mqtt = mqtt;
+
+    agent.pOtaInterface = &otaInterface;
+    pAgentCtx = &agent;
```
Comment (divekarshubham) [inline test/cbmc/proofs/requestJob_Mqtt/requestJob_Mqtt_harness.c:83]: Why do you have two agent context? Maybe do this:
`pAgentCtx->pOtaInterface = &otaInterface`

**Review by divekarshubham — CHANGES_REQUESTED** [review]

File: test/cbmc/proofs/requestJob_Mqtt/requestJob_Mqtt_harness.c [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+    /* OTA agent is defined globally in ota.c and thus cannot be NULL. */
+    __CPROVER_assume( pAgentCtx != NULL );
+
+    requestJob_Mqtt( pAgentCtx );
```
Comment (divekarshubham) [inline test/cbmc/proofs/requestJob_Mqtt/requestJob_Mqtt_harness.c:88]: Can we pass `&agent` here directly?

Comment (Bujain) [reply test/cbmc/proofs/requestJob_Mqtt/requestJob_Mqtt_harness.c]: Updated in fe1ecdf

File: test/cbmc/proofs/requestJob_Mqtt/requestJob_Mqtt_harness.c [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+{
+    OtaAgentContext_t * pAgentCtx;
+    OtaMqttInterface_t mqtt;
+    OtaAgentContext_t agent;
+    OtaInterfaces_t otaInterface;
+
+    /* publish reference to the mqtt function is expected to be assigned by the user and thus
+     * assumed not to be NULL. */
+    mqtt.publish = publish;
+    otaInterface.mqtt = mqtt;
```
[the remark this answers is not in the document]
Comment (Bujain) [reply test/cbmc/proofs/requestJob_Mqtt/requestJob_Mqtt_harness.c]: Updated in fe1ecdf

File: test/cbmc/proofs/requestJob_Mqtt/requestJob_Mqtt_harness.c [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+OtaMqttStatus_t __CPROVER_file_local_ota_mqtt_c_subscribeToJobNotificationTopics( const OtaAgentContext_t * pAgentCtx )
+{
+    OtaMqttStatus_t mqttStatus;
+
+    return mqttStatus;
+}
+
+OtaMqttStatus_t publish( const char * const pacTopic,
```
[the remark this answers is not in the document]
Comment (Bujain) [reply test/cbmc/proofs/requestJob_Mqtt/requestJob_Mqtt_harness.c]: Updated in fe1ecdf

File: test/cbmc/proofs/requestJob_Mqtt/requestJob_Mqtt_harness.c [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+                         uint16_t usTopicLen,
+                         const char * pcMsg,
+                         uint32_t ulMsgSize,
+                         uint8_t ucQoS )
+{
+    OtaMqttStatus_t status;
+
+    return status;
```
[the remark this answers is not in the document]
Comment (Bujain) [reply test/cbmc/proofs/requestJob_Mqtt/requestJob_Mqtt_harness.c]: Updated in fe1ecdf

File: test/cbmc/proofs/requestJob_Mqtt/Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (Bujain) [reply test/cbmc/proofs/requestJob_Mqtt/Makefile]: Updated in fe1ecdf

**Review by divekarshubham — APPROVED** [review]

**Review by aggarg — APPROVED** [review]

**Pull request merged**